### PR TITLE
Fixes up all files in the solution to use either tabs or spaces.

### PR DESCRIPTION
**Pull request by tannergooding**
This fixes up all files contained within the Roslyn.sln to use either tabs or spaces (but not both) according to the `.editorconfig` file (the only thing using tabs is sln files).

This does not necessarily cover all files in the repo.

### CODE REVIEW

**Comment by tannergooding** [comment]
FYI. @brettfo, @dotnet/roslyn

**Comment by TyOverby** [comment]
SHIP IT

**Comment by jaredpar** [comment]
Big change to look through.  Can we just answer a high level question?

Code is spaces correct?  

I don't care about other documents but code is spaces.

**Comment by tannergooding** [comment]
@jaredpar, that is what our config file says. I mostly just get annoyed when I open a random file and it says, this file contains both.

**Comment by JohnHamby** [comment]
We could probably write analyzers for C# and VB to flag tabs in whitespace.

**Comment by brettfo** [comment]
The diff shows you removed one line; please find and mark that.

File: src/Compilers/CSharp/Portable/Binder/Semantics/BestTypeInferrer.cs [hunk 1]
```diff
@@ -68,12 +68,12 @@ public static TypeSymbol InferBestType(ImmutableArray<BoundExpression> exprs, Co
...
         {
             // SPEC:    The second and third operands, x and y, of the ?: operator control the type of the conditional expression. 
-            // SPEC:    •	If x has type X and y has type Y then
-            // SPEC:        o	If an implicit conversion (§6.1) exists from X to Y, but not from Y to X, then Y is the type of the conditional expression.
-            // SPEC:        o	If an implicit conversion (§6.1) exists from Y to X, but not from X to Y, then X is the type of the conditional expression.
-            // SPEC:        o	Otherwise, no expression type can be determined, and a compile-time error occurs.
-            // SPEC:    •	If only one of x and y has a type, and both x and y, are implicitly convertible to that type, then that is the type of the conditional expression.
-            // SPEC:    •	Otherwise, no expression type can be determined, and a compile-time error occurs.
+            // SPEC:    •    If x has type X and y has type Y then
```
Comment (dpoeschl) [inline src/Compilers/CSharp/Portable/Binder/Semantics/BestTypeInferrer.cs]: These used to be aligned. I think it looks better this way, but just pointing out potentially unexpected consequences.

File: src/Compilers/CSharp/Portable/Parser/DocumentationCommentParser.cs [hunk 2]
```diff
@@ -859,23 +859,23 @@ protected override TNode WithAdditionalDiagnostics<TNode>(TNode node, params Dia
...
-        /// 					| _OperatorKeyword _OperatorToken (_OpenParenToken CrefParameterList? _CloseParenToken)?
-        /// 					| (_ImplicitKeyword | _ExplicitKeyword) _OperatorKeyword CrefParameterType (_OpenParenToken CrefParameterList? _CloseParenToken)?
-        /// CrefParameterList	= CrefParameter (_CommaToken CrefParameter)*
-        /// CrefParameter		= (_RefKeyword | _OutKeyword)? CrefParameterType
-        /// CrefParameterType	= CrefParameterType2 _QuestionToken? _AsteriskToken* (_OpenBracketToken _CommaToken* _CloseBracketToken)*
-        /// CrefParameterType2 	= (((_IdentifierToken _ColonColonToken)? CrefParameterType3) | _PredefinedTypeToken) (_DotToken CrefParameterType3)*
-        /// CrefParameterType3 	= _IdentifierToken (_LessThanToken CrefParameterType (_CommaToken CrefParameterType)* _GreaterThanToken)?
+        /// CrefType             = CrefFirstType (_DotToken CrefName)*
+        /// CrefMember             = CrefName (_OpenParenToken CrefParameterList? _CloseParenToken)?
+        ///                     | _ThisKeyword (_OpenBracketToken CrefParameterList _CloseBracketToken)?
```
Comment (dpoeschl) [inline src/Compilers/CSharp/Portable/Parser/DocumentationCommentParser.cs]: Derivations used to be aligned

**Comment by dpoeschl** [comment]
I'll be shocked (/disappointed) if no formatting unit tests fail after this.

**Comment by dpoeschl** [comment]
Or more terrifyingly, if the formatting tests still pass but for the wrong reason (and we lose coverage). /cc: @heejaechang @basoundr

File: src/Compilers/CSharp/Portable/Binder/Semantics/BestTypeInferrer.cs (hunk 1 above)
Comment (tannergooding) [inline src/Compilers/CSharp/Portable/Binder/Semantics/BestTypeInferrer.cs]: Fixed. I am going through and making sure things are still aligned everywhere.

**Comment by tannergooding** [comment]
Some formatter tests failed. Although, IMO, they were already broken. They should be hardcoding `tab` as `\t` if it is actually required, rather than relying on something that may be modified by an extension or checkout/checkin command.

**Comment by basoundr** [comment]
> if the formatting tests still pass but for the wrong reason (and we lose coverage)

This has definitely happened in the past where a bunch of code never ran because a different behavior introduced for a different reason somehow encapsulated the expected behavior, which has resulted in some code never getting to run

**Comment by CyrusNajmabadi** [comment]
Tests can't necessarily use \t as they may be using verbatim strings.  

The checkin argument is unconvincing because that's what we have code reviews fur. 

The rule isn't 'no tabs'. It's no tabs for indentation of our own code.

File: src/EditorFeatures/CSharpTest/PerfTests/Sources/CSharpGoldilocksInput-MultipliedDelay.txt [hunk 3]
```diff
@@ -1,1205 +1,1205 @@
-p	0
```
Comment (heejaechang) [inline src/EditorFeatures/CSharpTest/PerfTests/Sources/CSharpGoldilocksInput-MultipliedDelay.txt]: is it okay to change this file?

File: src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditStatementTests.cs [hunk 4]
```diff
@@ -5897,16 +5897,16 @@ void F()
...
-		return 1;
-	}
+    int Z(Func<int> f)
+    {
+        return 1;
+    }
 
     void F()
     {
-		var result = from a in new[] { 1 }
```
Comment (heejaechang) [inline src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditStatementTests.cs]: why is this even changing test code?

**Comment by heejaechang** [comment]
not sure why it is changing test snippet code. if you believe test code must use \t rather than actual tab inside of string literal. you should make that change rather than converting everything to spaces.

by the way, I don't think any tool will convert tab in string literal since that basically change semantic of code. unless the tool is specifically created to change meaning of the code in which case I don't believe user will go unnoticed since the user explicitly asked for it.

**Comment by heejaechang** [comment]
by the way, what formatting test not failed when it should by converting tab to space? can you show list those so that we can find out and fix those tests?

there are multiple kinds of tests on formatting. one of them is not touching code area which is not asked to format. I believe there should be some test that says make sure code in those area (such as tab) didn't convert to spaces (default behavior)

or there is test that says ignore whitespace changes to make sure formatting didn't introduce or remove tokens from code.

or there is test that says make sure tab is converted spaces.

so we need to know what kind of tests have failed to know what the underlying issue is.

...

by the way, I am +1 on normalizing our code to spaces

**Comment by tannergooding** [comment]
This was just a blanketed change. There is obviously still some need for cleanup (ensuring comment alignment remains consistent and ensuring tests still use tabs).

I will continue working on this in my spare time, but this at least gives some perspective on how big of a change is needed.

**Comment by CyrusNajmabadi** [comment]
Yeah, a blanket change concerns me.  I'd much rather have us just do something like run the .net formatter over our code.  Does it not already have a setting for converting all indentation to tabs or spaces?

**Comment by jaredpar** [comment]
No it doesn't have such a setting because it's really, really tricky to get correct.

**Pull request closed without merging**